compiler/hash-ast/src/ast.rs

```diff
@@ -912,33 +857,53 @@ pub enum ExpressionKind<'c> {
     Typed(TypedExpr<'c>),
     Block(BlockExpr<'c>),
     Import(ImportExpr<'c>),
+    StructDef(StructDef<'c>),
+    EnumDef(EnumDef<'c>),
+    Bound(Bound<'c>),
```

Comment:
Can this be renamed to `TypeFunctionDef`? Also, maybe `FunctionDef` should be here too, instead of `LiteralExpr::Function`?